src/opnsense/mvc/app/controllers/OPNsense/Firewall/Api/FilterController.php

```diff
@@ -221,6 +225,8 @@ public function searchRuleAction()
 
             if (empty($interfaces)) {
                 $is_if = empty($record['interface']) || count(explode(',', $record['interface'])) > 1;
+            } elseif ((string)$record['interfacenot'] === "1") {
+                $is_if = true;
```

Comment:
```suggestion
                /* XXX given that $interfaces is deduplicated multiple interfaces unmute the match */
                $is_if = empty($record['interface']) || count($interfaces > 1) || $record['interface'] != $interfaces[0];
```

Reply:
FWIW, I'd like to restructure the logic checks in this block since they are a mostly upside down by checking the same things in conditionals.